### ISSUE TITLE
Authorization Checking Service; cleanup

### ISSUE THREAD

**Issue by SimoneZaza** [cherrytwist/Coordination#1302]
## Description
We need an Authorization (Checking) Service that is horizontally scalable and stateless, so that it can enable extracting other microservices from the Server.

## Goal
We have enabled `FileService` to be extracted from the Alkemio Server.

## Hypothesis
Having more, smaller services will improve scalability and will result in deployments of smaller changes, thus enabling quicker deployments with lesser risk in production.

## Must have scope
 - Authorization (Engine) Library, re-used between a service and the server
 - Authorization Service usable in 3rd party services, e.g. FileService
 - Authorization Library used in the Alkemio Server
 - Horizontally scalable
 - DTOs, Typings re-used
 - The service should provide an endpoint **`isAccessGranted(authPolicyID, agentID, additionalRules)`** 

Plus
- [ ] Existing functionality for whiteboard collaboration + file serving moved out of the server
## Out-of-scope for v1
- Caching
- Messaging bus / patterns

## Stakeholders
Neil Smyth, Valentin Yanakiev, Svetoslav Petkov, Alkemio Team

## Implementation Details
 - Authorization Service to use the existing messaging infrastructure (for its first version)
 - Usage of `MessagePattern` (same as `Guidance Engine`, `Matrix Adapter`)
https://alkem.io/building-alkemio/challenges/technicaldesigns-9455/opportunities/robustservicemic-4127/collaboration/microservicesalke-9412

**Comment by hoyterutteman** [comment]
Same as 1602?

**Comment by hoyterutteman** [comment]
To have tactical brainstorm before we can close this one

**Comment by valentinyanakiev** [comment]
**Status reconciliation — 2026-06-16** (verified against live code + the acceptance/production clusters).

**Built & deployed.** `alkem-io/authorization-evaluation-service` (Go) is a mature, stateless service — evaluator engine, sqlc DB layer (read-only on the Alkemio DB: `credential` + `authorization_policy`), caching + invalidation, NATS **and** h2c transports, circuit-breaker/retry resilience, Prometheus metrics. Releases up to **v0.0.4**, which is running `1/1` in **both acceptance and production** (deployed 2026-04-29, 0 restarts since). `isAccessGranted` is served as an evaluate call: `{actorId, privilege, authorizationPolicyId} → {allowed, reason}`.

**Sub-issue reality:**

| Sub-issue | Board | Reality |
|---|---|---|
| #3881 Prototype | closed | ✅ done (well beyond a prototype) |
| #4039 Remove file/wb-collab from server | closed | ✅ done |
| #5529 Networking & infra | closed | ✅ done — base manifests + ServiceMonitor, in acc + prod overlays |
| #5528 Automated deployments | open | 🟢 effectively done — CI/CD + Dockerfile + releases v0.0.1→v0.0.4 all exist |
| #5545 File consumer | open | 🟡 file-service wired (h2c/NATS); missing parity tests in `test-suites` |
| #5530 Consumers | open | 🔴 only `file-service` + `wopi-service` consume it; `server`, whiteboard-collab and memos do not |

**Remaining must-have scope:** "Authorization Library used in the Alkemio Server" is **not** met — the server has not been migrated to the service. That, plus the remaining consumers (whiteboard-collaboration, memos) and the file-service parity tests, is the work left before this epic can close.

_(Service repo last touched 2026-04-29 — work appears paused, consistent with @hoyterutteman's "tactical brainstorm before we close" note.)_
